### PR TITLE
Local env fetching

### PR DESCRIPTION
**Pull request by MathieuGilbert**
(Optionally) retrieve and overwrite .env files from Gitlab locally before deploying. This gets you:

.env (same as .development)
.env.development
.env.staging
.env.production

### CODE REVIEW

File: lib/capistrano-deploy/environment.rb [hunk 1]
```diff
@@ -27,16 +25,25 @@ def self.load_into(configuration)
             task :verify do
               run "ssh-keyscan #{environment_host} | tee --append $HOME/.ssh/known_hosts"
             end
-
```
Comment (MathieuGilbert) [inline lib/capistrano-deploy/environment.rb:30]: From here up is just whitespace changes.

File: lib/capistrano-deploy/environment.rb [hunk 2]
```diff
@@ -27,16 +25,25 @@ def self.load_into(configuration)
...
           end
 
           namespace :local do
             desc "Fetch ENV file from remote git repository to your local machine"
             task :fetch_config do
-              raise "Not yet implemented"
+              if /^y/i =~ Capistrano::CLI.ui.ask("Do you wish to overwrite your local #{environment_file} file (y/n)? ")
+                temp_folder = './tmp/env_config'
```
Comment (ryanjones) [inline lib/capistrano-deploy/environment.rb:34]: I think it's safe to say that tmp will always exist.

File: lib/capistrano-deploy/environment.rb [hunk 3]
```diff
@@ -27,16 +25,25 @@ def self.load_into(configuration)
...
+
+                system "rm -rf #{temp_folder} && git clone -n #{environment_repository} --depth 1 #{temp_folder}"
+                system "cd #{temp_folder} && git checkout HEAD #{app_name}/.env.*"
```
Comment (ryanjones) [inline lib/capistrano-deploy/environment.rb:37]: checkout head only downloads the related app's files. Shweet.

**Comment by ryanjones** [comment]
:elephant: :elephant: :elephant: 

I think we need to think about the whole dl'ing of staging and prod in the future, but this is a good first step.

Can you open the PR's on my_ama/sent/gate to update the gem? And also send an email to the group about the changes?

**Pull request merged**